xonsh/completer.py

```diff
@@ -316,7 +318,12 @@ def bash_complete(self, prefix, line, begidx, endidx):
             out = ''
 
         space = ' '
-        rtn = {s + space if s[-1:].isalnum() else s for s in out.splitlines()}
+        slash = '/'
+        rtn = {_normpath(repr(s + (slash if os.path.isdir(s) else '')))
+               if COMPLETION_WRAP_TOKENS.intersection(s) else
+               s + space
+               if s[-1:].isalnum() else
```

Comment:
Maybe this whole expr should just be its own function, an then `rtn = set(map(f, out.splitlines()))`